### PR TITLE
Add some confetti when the player wins a match or a puzzle

### PR DESCRIPTION
**Pull request by alexandrepiveteau**
- [x] Show some confetti when the player wins a match
- [x] Show some confetti when the player wins a puzzle

## Demo

https://user-images.githubusercontent.com/6318990/168475680-9d33bcd4-fe64-42c9-bcc6-4241895813b4.mp4

### CODE REVIEW

File: mobile/src/main/java/ch/epfl/sdp/mobile/ui/game/Confetti.kt [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+    require(spread > 0f && spread <= 360f) { "Spread must be a valid angle." }
+    require(colors.isNotEmpty()) { "Must provide at least one confetti color." }
+
+    val firstAngle = angle.toDouble() - spread / 2f
+    val secondAngle = angle.toDouble() + spread / 2f
+    val fromAngle = minOf(firstAngle, secondAngle)
+    val untilAngle = maxOf(firstAngle, secondAngle)
+
+    val confettiAngle = Random.nextDouble(fromAngle, untilAngle).toFloat()
+    val confettiDistance = Random.nextDouble(from = 12.0, until = 36.0)
```
Comment (matt989253) [inline mobile/src/main/java/ch/epfl/sdp/mobile/ui/game/Confetti.kt:71]: I think these values should be extracted to constants, or maybe made as parameters with default values?

File: mobile/src/main/java/ch/epfl/sdp/mobile/ui/game/Confetti.kt [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+
+    val x = (cos(2 * PI * confettiAngle / 360f) * confettiDistance).dp
+    val y = (-sin(2 * PI * confettiAngle / 360f) * confettiDistance).dp
+
+    val state =
+        Confetti(
+            color = colors.random(),
+            initialSpeed = DpOffset(x * 30, y * 30),
+            targetSlide = DpOffset(0.dp, 180.dp),
```
Comment (matt989253) [inline mobile/src/main/java/ch/epfl/sdp/mobile/ui/game/Confetti.kt:80]: Here as well.

**Review by matt989253 — APPROVED** [review]
🎊🎊🎊🎊

File: mobile/src/main/java/ch/epfl/sdp/mobile/ui/game/Confetti.kt (hunk 1 above)
Comment (alexandrepiveteau) [reply mobile/src/main/java/ch/epfl/sdp/mobile/ui/game/Confetti.kt]: Extracted to default params in 7e5b878c4117c72514b5d7c617ff6c7fa511c0b1

File: mobile/src/main/java/ch/epfl/sdp/mobile/ui/game/Confetti.kt (hunk 2 above)
Comment (alexandrepiveteau) [reply mobile/src/main/java/ch/epfl/sdp/mobile/ui/game/Confetti.kt]: Extracted to default params in 7e5b878c4117c72514b5d7c617ff6c7fa511c0b1

**Review by matt989253 — APPROVED** [review]

**Pull request merged**